lib/document/fetchDocumentsFactory.js

```diff
@@ -23,7 +23,12 @@ function fetchDocumentsFactory(
    * @returns {Promise<Document[]>}
    */
   async function fetchDocuments(contractId, type, options, dbTransaction = undefined) {
-    const documentRepository = await createDocumentRepository(contractId, type);
+    let documentRepository;
+    try {
+      documentRepository = await createDocumentRepository(contractId, type);
+    } catch (error) {
+      throw new InvalidQueryError([error]);
```

Comment:
We should check for specific error otherwise any error will lead to InvalidQueryError